### DISCUSSION TITLE
How to create molecule image similar with molconvert using RDKit

### DISCUSSION THREAD

**Discussion by ewijaya** [rdkit/rdkit#4649] [General] [answered]
I have the following molecule in SMILE format (saved as `my_molecule.smi`):

    [H]N[C@@H](CCCCN)C(=O)N[C@@H](CCCCN)C(=O)N[C@@H](C)C(=O)N[C@@H](CC1=CNC2=C1C=CC=C2)C(O)=O

I created an image using ChemAxon Marvin's [**molconvert**][1] command:

    MarvinSuite/bin/molconvert "jpeg:w800:h800" my_molecule.smi -o mol.jpg

It produces the following image:

[![enter image description here][2]][2]

Then I use [RDKit][3], with the following code:

    from rdkit import Chem
    from rdkit.Chem.Draw import IPythonConsole
    m = Chem.MolFromSmiles('[H]N[C@@H](CCCCN)C(=O)N[C@@H](CCCCN)C(=O)N[C@@H](C)C(=O)N[C@@H](CC1=CNC2=C1C=CC=C2)C(O)=O')
    m

I get this image:

[![enter image description here][4]][4]

Notice that it is circular, and missing an H3C.
With RDKit, how can I create the identical image with the one made by `molconvert`?
i.e. straight format and H3C included.

Python code can be tried in [**my Google Colab**][5].

  [1]: https://docs.chemaxon.com/display/docs/molconvert.md
  [2]: https://i.stack.imgur.com/Wy8Mz.jpg
  [3]: https://www.rdkit.org
  [4]: https://i.stack.imgur.com/7goIX.png
  [5]: https://colab.research.google.com/drive/1AJy1YcIpSvr2x5s-LFHisKU4LhI0cOEW?usp=sharing

**Comment by dehaenw** [comment, marked as answer]
hello @ewijaya.
those are the same two molecules, whether you prefer the linear drawing or the more square one is only a question of taste. In any case, there is a somewhat hacky way to get the structure to look more linear, using GenerateDepictionMatching2DStructure to align to the linear substructure (maybe there is some better way, because this is not general). As for drawing the methyl as CH3 explicitly, this is likewise just a question of taste. One solution is to add CH3 as an abbreviation using rdAbbreviations.
```python
from rdkit import Chem
from rdkit.Chem import rdAbbreviations,AllChem
abbrevs = rdAbbreviations.ParseAbbreviations("CH<sub>3</sub> [Ch3]")
m = Chem.MolFromSmiles('[H]N[C@@H](CCCCN)C(=O)N[C@@H](CCCCN)C(=O)N[C@@H](C([H])([H])[H])C(=O)N[C@@H](CC1=CNC2=C1C=CC=C2)C(O)=O')
m = rdAbbreviations.CondenseMolAbbreviations(m,abbrevs)
template= Chem.MolFromSmiles("C(=O)CNC(=O)CNC(=O)CNC(=O)CCCC")
AllChem.Compute2DCoords(template)
AllChem.GenerateDepictionMatching2DStructure(m,template)
display(m)
```
outputs:
![image](https://user-images.githubusercontent.com/66372095/139310968-206f903e-3cdf-4477-be73-9970c35e8e75.png)

**Reply by ewijaya** [reply]
Thank you. It works.

**Reply by DavidACosgrove** [reply]
An easier way to achieve the aim of showing a methyl as CH3 is with the drawingOptions:

from rdkit.Chem import AllChem
from rdkit.Chem import Draw
from rdkit.Chem.Draw import rdMolDraw2D

```
m = AllChem.MolFromSmiles('[H]N[C@@H](CCCCN)C(=O)N[C@@H](CCCCN)C(=O)N[C@@H](C([H])([H])[H])C(=O)N[C@@H](CC1=CNC2=C1C=CC=C2)C(O)=O')
drawer = rdMolDraw2D.MolDraw2DCairo(300, 300)
drawer.drawOptions().explicitMethyl = True
rdMolDraw2D.PrepareAndDrawMolecule(drawer, m)
drawer.FinishDrawing()
with open('with_methyls.png', 'wb') as f:
    f.write(drawer.GetDrawingText())
```
![image](https://user-images.githubusercontent.com/9198870/139462917-ea56c6d0-ab53-465f-9ecf-82e90e716b1f.png)

I'm sorry I'm old-school enough to write code in emacs rather than iPython, and so write pictures to a file.
HTH,
Dave
